src/services/userService.js

```diff
@@ -0,0 +1,91 @@
+const { userDao } = require('../models');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const { throwError } = require('../utils/throwError');
+
+const findUser = async (userId) => {
+  return await userDao.findById(userId);
+};
+
+const signUp = async (
+  name,
+  email,
+  password,
+  phoneNumber,
+  zipCode,
+  address,
+  addressDetails
+) => {
+  const emailRegx = /^([a-zA-Z0-9._%-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})$/;
+  if (!email.match(emailRegx)) throwError(400, 'INVALID_EMAIL');
+
+  const passwordRegx = /^(?=.*[0-9])(?=.*[a-z])(?=.*\W)(?!.* ).{8,16}$/;
+  if (!password.match(passwordRegx)) throwError(400, 'INVALID_PASSWORD');
+
+  const saltRounds = 12;
+  const bcryptPassword = await bcrypt.hash(password, saltRounds);
+
+  await userDao.signUp(
+    name,
+    email,
+    bcryptPassword,
+    phoneNumber,
+    zipCode,
+    address,
+    addressDetails
+  );
+};
+
+const duplicateEmail = async (email) => {
+  const existingUser = await userDao.findByEmail(email);
+
+  if (existingUser) throwError(400, 'DUPLICATED_EMAIL_ADDRESS');
+};
+
+const signIn = async (email, password) => {
+  const existingUser = await userDao.findByEmail(email);
+  if (!existingUser) throwError(400, 'USER_NOT_FOUND');
+  let isSeller = false;
+  if (existingUser.seller_id != null) isSeller = true;
+
+  const checkPassword = await bcrypt.compare(password, existingUser.password);
+  if (!checkPassword) throwError(400, 'WRONG_PASSWORD');
+
+  const token = jwt.sign({ id: existingUser.id }, process.env.JWT_SECRET);
+
+  return { accessToken: token, isSeller: isSeller };
+};
+
+const sellerSignUp = async (
+  name,
+  image,
+  zipCode,
+  address,
+  addressDetails,
+  phoneNumber,
+  userId
+) => {
+  const existingSeller = await userDao.findSeller(name);
+  if (existingSeller) throwError(400, 'INVALID_NAME');
+
+  const existingUser = await userDao.findById(userId);
+  if (existingUser.seller_id !== null) throwError(400, 'ALREADY_SELLER');
+
```

Comment:
```suggestion
  const existingUser = await userDao.findById(userId);
  if (existingUser.seller_id !== null) throwError(400, 'ALREADY_SELLER');

  const existingSeller = await userDao.findSeller(name);
  if (existingSeller) throwError(400, 'INVALID_NAME');
  
```
1. if(existing.User.seller_id !== null) 인 조건과 if(existingSeller)인 조건이 동일한 조건이 아닌가요? 
2. 만약 아니라면 하나는 이미 셀러인지를 확인하는 과정이고, 하나는 셀러 이름이 중복되면 안된다는 조건인 것 같은데 맞을까요?
3. 2번이 맞다면, ALREADY_SELLER인지를 확인하는 과정이 더 먼저 있어야 맞을 것 같습니다. 해당 에러가 제대로 잡히나요?? 아래 두가지 이유 때문에 순서 바뀌는게 좋을 것 같은데, 제가 비즈니스상 로직 잘못 이해하고 있으면 알려주세요! 
3-1. 이미 셀러인 사람이 동일한 이름으로 셀러 등록을 하려면 'INVALID_NAME'이 뜨는 로직인데, 'ALREADY_SELLER'가 떠야 할 것 같습니다. 유효한 셀러 이름인지 판단하는 과정보다는 그 사람이 셀러를 등록할 수 있는 상황인지 판단하는 과정이 더 상위개념이라고 판단됩니다.
3-2. 이미 ALREADY_SELLER라면 그 이후에 유효한 셀러명인지 여부를 판단하는 과정이 아예 불필요하다고 생각됩니다. SELECT query문 1개만 사용해도 되는 상황, user/seller 각각 SELECT 문을 2개 쓰게 되네요 :)

Reply:
2번이 맞습니다! 
두 가지를 분리해 둔 이유는 셀러들끼리 이름이 안겹치게 하고, 이미 셀러 아이디가 있으면 셀러 등록을 할 수 없게끔 하려고 분리해 두었습니다!

그럼 셀러 이름을 받는 로직을 수정하면 될까요?

Reply:
아 그럼 코드의 순서만 바꾸면 될 것 같아요! if문의 순서가
1. 현재 셀러인지 먼저 확인하고 -> 이미 셀러면 Error
2. 아직 셀러가 아니라면 셀러 이름 가능한지 체크해서 -> 겹치는 이름이면 Error
이런 순서로 진행되어야 할 것 같습니다.

Reply:
넵!